packages/core/src/index.ts

```diff
@@ -72,44 +77,49 @@ export function buildSendMail<T>(options: BuildSendMailOptions<T>) {
   }
 
   return async function sendMail(mail: ComponentMail) {
-    const forcePreview =
-      mail.forcePreview ||
-      (process.env.NODE_ENV !== "production" && !mail.dangerouslyForceDeliver);
-
     if (!mail.html && typeof mail.component === "undefined")
       throw new Error("sendMail requires either html or a component");
 
-    const { html, errors } =
-      mail.html || !mail.component
-        ? { html: mail.html, errors: [] }
-        : render(mail.component);
-
-    if (errors?.length) {
-      error(errors);
-      throw new Error(errors.join(";"));
+    const { NODE_ENV, MAILING_API_URL, MAILING_API_KEY, MAILING_DATABASE_URL } =
+      process.env;
+    const {
+      component,
+      dangerouslyForceDeliver,
+      templateName,
+      previewName,
+      forcePreview,
+      ...mailOptions
+    } = mail;
+    mailOptions as SendMailOptions;
+    mailOptions.from ||= options.defaultFrom;
+
+    const previewMode =
+      !options.forceNotTestMode &&
+      (forcePreview || (NODE_ENV !== "production" && !dangerouslyForceDeliver));
+
+    // Do not send emails analytics if MAILING_DATABASE_URL is set
+    // this means sendMail is being called from the REST API and analytics will be handled there
+    const analyticsEnabled =
+      !MAILING_DATABASE_URL && MAILING_API_URL && MAILING_API_KEY;
+
+    // Get html from the rendered component if not provided
+    let derivedTemplateName;
+    if (component && !mailOptions.html) {
+      const { html: renderedHtml, errors } = render(component);
+      if (errors?.length) {
+        error(errors);
+        throw new Error(errors.join(";"));
+      }
+      derivedTemplateName = component.type.name;
```

Comment:
Hmmm. This is why we were going to lint.

Still feels sketchy because we re;y on the filename for all of the template naming in mailing cli (e.g. for the file tree and template names in the moduleManifest), but here we use the function name which can be different. Trying to figure out if there's a better way to do this. Ideally we'd use the filename here somehow.

Reply:
re-opened #284

Reply:
https://github.com/sofn-xyz/mailing/pull/287